crates/re_viewer/src/ui/override_ui.rs

```diff
@@ -0,0 +1,266 @@
+                    row.col(|ui| {
+                        if ctx
+                            .re_ui
+                            .small_icon_button(ui, &re_ui::icons::CLOSE)
+                            .clicked()
+                        {
+                            // Note: need to use the blueprint store since the data might
+                            // not exist in the recording store.
+                            ctx.save_empty_blueprint_component_name(
+                                ctx.store_context.blueprint.store(),
+                                &overrides.override_path,
+                                *component_name,
+                            );
+                        }
+                    });
+                    // Component label
+                    row.col(|ui| {
+                        temporary_style_ui_for_component(ui, component_name, |ui| {
+                            item_ui::component_path_button(
+                                ctx,
+                                ui,
+                                &ComponentPath::new(entity_path.clone(), *component_name),
+                            );
+                        });
+                    });
+                    // Editor last to take up remainder of space
+                    row.col(|ui| {
+                        let component_data = match store_kind {
+                            StoreKind::Blueprint => {
+                                let store = ctx.store_context.blueprint.store();
+                                let query = ctx.blueprint_query;
+                                get_component_with_instances(
+                                    store,
+                                    query,
+                                    entity_path,
+                                    *component_name,
+                                )
+                            }
+                            StoreKind::Recording => get_component_with_instances(
+                                store,
+                                &query,
+                                entity_path,
+                                *component_name,
+                            ),
+                        };
+
+                        if let Some((_, _, component_data)) = component_data {
+                            ctx.component_ui_registry.edit(
+                                ctx,
+                                ui,
+                                UiVerbosity::Small,
+                                &query,
+                                store,
+                                entity_path,
+                                &overrides.override_path,
+                                &component_data,
+                                instance_key,
+                            );
+                        } else {
+                            // TODO(jleibs): This shouldn't happen. Warn instead?
+                            ui.weak("(empty)");
```

Comment:
we have `re_ui.warning_text` for cases like this - this could then print out what's happening, even with a longer explanation in the tooltip

Reply:
I'm not totally confident it should be a warn though either. Changing the comment.